src/puya/ir/context.py

```diff
@@ -114,16 +114,11 @@ class IRBuildContextWithFallback(IRBuildContext):
     @contextlib.contextmanager
     def log_exceptions(self, fallback_location: SourceLocation | None = None) -> Iterator[None]:
         fallback_location = fallback_location or self.default_fallback
-        try:
-            yield
-        except CodeError as ex:
-            self.errors.error(str(ex), location=ex.location or fallback_location)
-        except InternalError as ex:
-            self.errors.error(f"FATAL {ex!s}", location=ex.location or fallback_location)
-            crash_report(ex.location or fallback_location)
-        except Exception as ex:
-            self.errors.error(f"UNEXPECTED {ex!s}", location=fallback_location)
-            crash_report(fallback_location)
+        with log_exceptions(self.errors, fallback_location):
+            try:
+                yield
+            finally:
+                pass
```

Comment:
The with/finally here is redundant, just the yield should be the same.
Can we just return the generator?

Reply:
Fixed